ui/page/main_page.go

```diff
@@ -400,14 +400,20 @@ func (mp *MainPage) HandleUserInteractions() {
 				if mp.receivePage == nil {
 					mp.receivePage = NewReceivePage(mp.Load)
 				}
-				pg = mp.receivePage
+				pg = NewReceivePage(mp.Load)
 			}
 
 			if pg.ID() == mp.currentPageID() {
 				continue
 			}
 
-			mp.ChangeFragment(pg)
+			if !mp.WL.MultiWallet.IsConnectedToDecredNetwork() {
```

Comment:
Implement this logic on the individual pages and not here
Currently send page doesn't permit sending when godcr isn't connected to the network. So no point implementing that logic.

I will suggest that the receive page should let users generate address after account discovery for imported accounts and at any point for other accounts created on the app.

Reply:
This raises a privacy concern in the sense that addresses might be reused when a wallet is not synced to the tip as the information available is not sufficient to determine if an address has been used before.

Reply:
Should be best to balance security and usability. If a user tries to retrieve an address when the wallet isn't fully synced the user should get a notification about the security implications, but should be able to retrieve an address irrespective. In some cases users will be more concerned about not waiting to sync than reusing an address.

Reply:
The receive page should not be accessible until wallets are full my synced… you can test the mobile android or iOS for more information

Reply:
For the first point you made, if the send page isn't usuable, then might as well prevent them from accessing it and letting them know why, seams like a cleaner approach

Then, while I get your point that it could be frustrating to wait for sync before you can get a receiving address, even if you get an address before sync completes, sending funds there wouldn't reflect until sync is done, so they might as well wait